plugins/funind/g_indfun.mlg

```diff
@@ -97,13 +97,25 @@ let functional_induction b c x pat =
 
 TACTIC EXTEND newfunind
 | ["functional" "induction" ne_constr_list(cl) fun_ind_using(princl) with_names(pat)] ->
-     {
-       let c = match cl with
-         | [] -> assert false
-         | [c] -> c
-         | c::cl -> EConstr.applist(c,cl)
-       in
-       Extratactics.onSomeWithHoles (fun x -> functional_induction true c x pat) princl }
+   {
+     match cl with
+     | [] -> assert false
+     | [c] ->
+        Extratactics.onSomeWithHoles
+          (fun x -> functional_induction true c x pat) princl
+     | c::cl ->
+        let appl = EConstr.applist(c,cl) in
+        let check =
+          Proofview.Goal.enter begin fun gl ->
+            let env = Tacmach.New.pf_env gl in
+            let sigma = Tacmach.New.project gl in
+            let _,_ = Typing.type_of env sigma appl in
+            Tacticals.New.tclIDTAC
+            end in
```

Comment:
this build an idtac that performs the type checking. There must be a cleaner way...